src/components/ThemeToggle/ThemeToggle.tsx

```diff
@@ -6,34 +6,53 @@ import styles from './ThemeToggle.module.scss'
 import { useAnalytics } from 'stores/analyticsContext'
 import { useUser } from 'hooks/useUser'
 import { useEditThemeMutation } from 'operations/portal/mutations/editTheme.g'
+import { SessionUser } from 'types'
+import { useGetThemeQuery } from 'operations/portal/queries/getTheme.g'
 
 const ThemeToggle = ({ flags }: { flags?: LDFlagSet }) => {
+  const { data } = useGetThemeQuery()
   const { theme, setTheme } = useTheme()
   const { trackEvent } = useAnalytics()
   const { user } = useUser()
   const [handleEditThemeMutation] = useEditThemeMutation()
 
   useEffect(() => {
-    if (user && theme) {
+    if (data && theme && data.theme !== theme) {
+      // If there is a discrepancy between what is in the db and what is
+      // in local storage, then use what is in local storage.
+      setTheme(theme)
+    } else if (data) {
+      setTheme(data.theme)
+    }
+  }, [])
+
+  const handleThemeChangeAndTracking = (
+    user: SessionUser | null,
+    newTheme: string
+  ) => {
+    trackEvent(
+      'Dark mode',
+      'Click on light/dark mode toggle',
+      'Light/Dark mode toggle'
+    )
+
+    if (user) {
       handleEditThemeMutation({
         variables: {
           userId: user.userId,
-          theme: theme,
+          theme: newTheme,
         },
       })
     }
-  }, [user, theme])
+  }
 
   return flags && flags.darkModeToggle ? (
     <button
       type="button"
       onClick={() => {
-        trackEvent(
-          'Dark mode',
-          'Click on light/dark mode toggle',
-          'Light/Dark mode toggle'
-        )
-        setTheme(theme === 'light' ? 'dark' : 'light')
+        const newTheme = theme === 'light' ? 'dark' : 'light'
+        setTheme(newTheme)
+        if (theme) handleThemeChangeAndTracking(user, newTheme)
```

Comment:
Question: When is it possible for `theme` to be false? Do we need to check it since `newTheme` will default to light and this only happens when a user is clicking?

Reply:
I initially left this if statement because I was getting a type error otherwise. I think it was something about `theme` potentially being null, but I've removed it and am no longer seeing the error.